features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -46,6 +48,15 @@ static nsapi_error_t mbed_lwip_err_remap(err_t err);
     #define MBED_NETIF_INIT_FN eth_arch_enetif_init
 #endif
 
+#ifndef LWIP_SOCKET_MAX_MEMBERSHIPS
+    #define LWIP_SOCKET_MAX_MEMBERSHIPS 4
+#endif
+
+#define next_registered_multicast_member(registry, index)       while(!(registry & (0x0001 << index))) { index++; }
```

Comment:
Magic macros seem a bit tricksy, and not gaining much here.

Maybe avoid confusion by making them static inline functions that return an updated index, rather than modifying index by value?

We assume C99 or later, so `static inline` is portable (and works the same as in C++03).